packages/financial-templates-lib/src/logger/SlackTransport.ts

```diff
@@ -45,26 +45,32 @@ interface DividerBlock {
 
 type Block = SectionBlock | DividerBlock; // Add more | types here to add more types of blocks.
 
-interface SlackFormatterResponse {
+type SlackFormatterResponse = {
   blocks: Block[];
-}
+};
 
 export const SLACK_MAX_CHAR_LIMIT = 3000;
+let botIdentifier: string | undefined;
 
 // Note: info is any because it comes directly from winston.
 function slackFormatter(info: any): SlackFormatterResponse {
   try {
     if (!("level" in info) || !("at" in info) || !("message" in info))
       throw new Error("WINSTON MESSAGE INCORRECTLY CONFIGURED");
 
+    botIdentifier ??= info["bot-identifier"];
+
     // Each part of the slack response is a separate block with markdown text within it.
     // All slack responses start with the heading level and where the message came from.
     const formattedResponse: SlackFormatterResponse = {
       // If the bot contains an identifier flag it should be included in the heading.
       blocks: [
         {
           type: "section",
-          text: { type: "mrkdwn", text: `[${info.level}] *${info["bot-identifier"]}* (${info.at})⭢${info.message}\n` },
+          text: {
+            type: "mrkdwn",
+            text: `[${info.level}] *${info["bot-identifier"] ?? botIdentifier}* (${info.at})⭢${info.message}\n`,
```

Comment:
Why'd you make this change?

Reply:
The log transport doesn't know what the `bot-identifier` is, so when it generates its own internal log message, `bot-identifier` reverts to `undefined`. This is unhelpful when trying to understand which bot emitted a bad log.